h2o-core/src/main/java/hex/grid/GridSearch.java

```diff
@@ -122,6 +124,8 @@ private GridSearch(Key<Grid> gkey, HyperSpaceWalker<MP, ?> hyperSpaceWalker, fin
       while (it.hasNext(model) && (it.max_models() > 0 && count++ < it.max_models())) { //only walk the first max_models models, if specified
         try {
           Model.Parameters parms = it.nextModelParameters(model);
+          // When filtering is in the game, `hasNext()` can return true but `it.nextModelParameters()` can skip this next permutation by returning null
```

Comment:
There is a desire to incorporate this case into `hasNext()` method of a `FIlteredHyperSpaceIterator` but to make a filtering decision we would need next `Model.Parameters` instance (which is not available from the interface method's signature)